pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -166,19 +161,30 @@ func (ctrl *VMRestoreController) updateVMRestore(vmRestoreIn *snapshotv1.Virtual
 					return 0, ctrl.doUpdateError(vmRestoreIn, err)
 				}
 
-				ctrl.Recorder.Eventf(
-					vmRestoreOut,
-					corev1.EventTypeNormal,
-					restoreCompleteEvent,
-					"Successfully completed VirtualMachineRestore %s",
-					vmRestoreOut.Name,
-				)
-
-				t := true
-				vmRestoreOut.Status.Complete = &t
-				vmRestoreOut.Status.RestoreTime = currentTime()
-				updateRestoreCondition(vmRestoreOut, newProgressingCondition(corev1.ConditionFalse, "Operation complete"))
-				updateRestoreCondition(vmRestoreOut, newReadyCondition(corev1.ConditionTrue, "Operation complete"))
+				updated, err := target.UpdateDoneRestore()
+				if err != nil {
+					logger.Reason(err).Error("Error updating done restore")
+					return 0, ctrl.doUpdateError(vmRestoreIn, err)
+				}
+
+				if !updated {
+					ctrl.Recorder.Eventf(
```

Comment:
consider extracting to 'completeRestore' function or something like that

Reply:
Why?  The indentation?  I personally think it is more readable inline.  Especially since the function would not be called in multiple spots